src/uniforms/sampler.rs

```diff
@@ -140,6 +140,36 @@ impl ToGlEnum for DepthTextureComparison {
     }
 }
 
+
+/// The border color to use for `SamplerWrapFunction::BorderClamp`.
+///
+/// This type encapsulates `[f32; 4]` internally as `[u32; 4]` so that
+/// it implements Hash and Eq. The conversions to and from `[f32; 4]`
+/// are provided by the `From` trait.
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub struct BorderColor([u32; 4]);
+
+impl From<[f32; 4]> for BorderColor {
+    #[inline]
+    fn from(value: [f32; 4]) -> BorderColor {
+        BorderColor(unsafe { std::mem::transmute(value) })
+    }
+}
+
+impl From<BorderColor> for [f32; 4] {
+    #[inline]
+    fn from(value: BorderColor) -> [f32; 4] {
+        unsafe { std::mem::transmute(value.0) }
+    }
+}
+
+impl BorderColor {
+    #[inline]
+    pub(crate) fn as_f32_array(&self) -> &[f32; 4] {
+        unsafe { std::mem::transmute(&self.0) }
```

Comment:
same here

Reply:
it seems that this one is not applicable to the array.map + f32::from_bits trick. any suggestions?

Reply:
ah yeah, good point. let's leave it as is.